server/src/main/java/io/crate/user/Privileges.java

```diff
@@ -114,17 +115,27 @@ public static void ensureUserHasPrivilege(Privilege.Clazz clazz,
         }
     }
 
-    private static boolean isInformationSchema(Privilege.Clazz clazz, @Nullable String ident) {
+    private static String getTargetSchema(Privilege.Clazz clazz, @Nullable String ident) {
+        String schemaName = null;
         if (Privilege.Clazz.CLUSTER.equals(clazz)) {
-            return false;
+            return schemaName;
         }
         assert ident != null : "ident must not be null if privilege class is not 'CLUSTER'";
-        String schemaName;
         if (Privilege.Clazz.TABLE.equals(clazz)) {
             schemaName = new IndexParts(ident).getSchema();
         } else {
             schemaName = ident;
         }
-        return InformationSchemaInfo.NAME.equals(schemaName);
+        return schemaName;
+    }
+
+    private static boolean isInformationSchema(Privilege.Clazz clazz, String ident) {
+        String targetSchema = getTargetSchema(clazz, ident);
+        return targetSchema != null && targetSchema.equals(InformationSchemaInfo.NAME);
```

Comment:
```suggestion
        return InformationSchemaInfo.NAME.equals(targetSchema);
```

Could turn the equals order around to remove the null check.